Extensions/CLIExtensionTemplate.s4ext

```diff
@@ -42,3 +42,6 @@ screenshoturls http://www.example.com/Slicer/Extensions/CLIExtensionTemplate/Scr
 
 # 0 or 1: Define if the extension should be enabled after its installation.
 enabled 1
+
+# Space separated list of tags indicating build system features that are required to build this extension
```

Comment:
If tag is really "tags indicating build system features" then the name should reflect that. It could be something like buildSystemRequiredFeatures. In the future we may want to split extensions to group extensions for other reasons than required build system features (e.g., simply to distribute the load on multiple computers), so a bit more generic name, such as buildGroups could be better.

Reply:
I agree. I think it would also be helpful to the developers if the set of "tags" (or whatever is the name) was standardized for the overall application curated in some central location, which would then be referenced from here. I can see that the way you organized it at the moment will serve your purposes, but if I were to leverage this mechanism, it would not be clear how to get started.

Reply:
That's up to us really. The idea of tags could be multipurpose, or it could be specific just to this build system check. Let me know what you all want and we can go that path.

Reply:
I've just realized that this information does not have to be in the s4ext file. As you see, it is really complicated to add anything at that level, it is still very limited (there is a hardcoded very simple logic that compares tags in the extension with tags configured on the system; you cannot implement complex logic or compiler checks in CMake).

Instead, at configure time the extension could detect automatically that for example cuda is not present (or some CMake or environment variables are not set), and generate an empty (or some special project) project. The extension build system would either detect that it is a skipped project and would not submit anything to the dashboard, or it would submit an almost empty record (it would mean more items on the dashboard, but there would be information about why the extension was not built).

Reply:
So, an extension developer would have to make a different find_package call? This moves the onus on extension developers to know which packages are "special".

Reply:
Or would the build system have to analyze an extension's CMakeLists for known special cases?

Reply:
For me, it doesn't seem that complicated to add a field to the s4ext file, and it's easy for the build system to parse the needs of the extension from the list. 

This change also readily integrates into the existing build system and requires no changes to extension CMake code.

Reply:
> Or would the build system have to analyze an extension's CMakeLists for known special cases?

OK, maybe the simple label comparison mechanism is good enough, and it is simpler to understand the concept.